src/main/java/redis/clients/jedis/CommandObjects.java

```diff
@@ -63,22 +63,16 @@ protected CommandArguments commandArguments(ProtocolCommand command) {
     return new CommandArguments(command);
   }
 
-  private final CommandObject<String> PING_COMMAND_OBJECT = new CommandObject<>(commandArguments(PING), BuilderFactory.STRING);
-
   public final CommandObject<String> ping() {
-    return PING_COMMAND_OBJECT;
+    return new CommandObject<>(commandArguments(PING), BuilderFactory.STRING);
```

Comment:
Why change this and following ones? Prefixed or not, keys won't change the command arguments.

Reply:
The original problem was that the inline objects like `PING_COMMAND_OBJECT` were calling `commandArguments` which eventually called the constructor of `CommandArgumentWithPrefixedKeys` before the constructor of `CommandObjectsWithPrefixedKeys` even ran. At this point `prefixString` is null and the constructor of `CommandObjectsWithPrefixedKeys` throws an exception when it calls `SafeEncode` on a null value.

The fundamental issue seems to be the use of an overridable `protected` method inside the superclass constructor, which can cause problems because it's not possible to know if the class has actually finished constructing from the superclass.

However, I believe we can work around this adding some logic that takes into account the possibility of `prefixString` being null inside `CommandArgumentsWithPrefixedKeys`, if it is desirable to keep the cached `CommandObject` values.